src/sfizz/Wavetables.cpp

```diff
@@ -307,4 +332,65 @@ const WavetableMulti* WavetablePool::getWaveSquare()
     return &wave;
 }
 
+const WavetableMulti* WavetablePool::getFileWave(const std::string& filename)
+{
+    auto it = _fileWaves.find(filename);
+    if (it == _fileWaves.end())
+        return nullptr;
+
+    return it->second.get();
+}
+
+void WavetablePool::clearFileWaves()
+{
+    _fileWaves.clear();
+}
+
+const WavetableMulti* WavetablePool::createFileWave(FilePool& filePool, const std::string& filename)
+{
+    if (const WavetableMulti* wave = getFileWave(filename))
+        return wave;
+
+    if (!filePool.preloadFile(filename, 0))
```

Comment:
I had this idea to use the same file loader as used for everything.
This loader can have an concept of caching, and it may interact directly in the future with the idea of global storage.

Reply:
I guess the Wavepool is also static now? The Filepool does not store these wavetables, it may store the beginning of files but this should be pretty moot in the end since we're mostly interested in the files and not the table. Most of the Filepool logic is centered on this preloading/streaming/oversampling complexity. Reading a file with libsndfile is maybe 3 lines of code on the other hand. What we may do is not compute a table if the static/global wavepool finds that it already has the proper table.

One thing that we may do is provide an API for file reading but not preloading in the Filepool, possibly with caching?

Reply:
What I thought of and forgot mentioning, it's also something that will check file changes and reload accordingly. I remember this happen for SFZ files but for samples I was not sure if it's the case or not.

Reply:
It happens for sfz files but the library users have to call it. Basically there's a "check if sfz files have changed" method and you have to call it and reload in your own time. In this case though, everything is purged and reloaded, at least for now. And this only checks the sfz files, not the sample files. If you change the sample files between preloading and actual loading, there will be pain...